cli/src/pcluster/models/cluster.py

```diff
@@ -220,23 +219,23 @@
         storage_type: str = None,
     ):
         super().__init__(mount_dir=mount_dir, shared_storage_type=SharedStorage.Type.FSX)
-        self.storage_capacity = Param(storage_capacity)
-        self.storage_type = Param(storage_type)
-        self.deployment_type = Param(deployment_type)
-        self.export_path = Param(export_path)
-        self.import_path = Param(import_path)
-        self.imported_file_chunk_size = Param(imported_file_chunk_size)
-        self.weekly_maintenance_start_time = Param(weekly_maintenance_start_time)
-        self.automatic_backup_retention_days = Param(automatic_backup_retention_days)
-        self.copy_tags_to_backups = Param(copy_tags_to_backups)
-        self.daily_automatic_backup_start_time = Param(daily_automatic_backup_start_time)
-        self.per_unit_storage_throughput = Param(per_unit_storage_throughput)
-        self.backup_id = Param(backup_id)
-        self.kms_key_id = Param(kms_key_id)
-        self.file_system_id = Param(file_system_id)
-        self.auto_import_policy = Param(auto_import_policy)
-        self.drive_cache_type = Param(drive_cache_type)
-        self.storage_type = Param(storage_type)
+        self.storage_capacity = Resource.init_param(storage_capacity)
+        self.storage_type = Resource.init_param(storage_type)
+        self.deployment_type = Resource.init_param(deployment_type)
+        self.export_path = Resource.init_param(export_path)
+        self.import_path = Resource.init_param(import_path)
+        self.imported_file_chunk_size = Resource.init_param(imported_file_chunk_size)
+        self.weekly_maintenance_start_time = Resource.init_param(weekly_maintenance_start_time)
+        self.automatic_backup_retention_days = Resource.init_param(automatic_backup_retention_days)
+        self.copy_tags_to_backups = Resource.init_param(copy_tags_to_backups)
+        self.daily_automatic_backup_start_time = Resource.init_param(daily_automatic_backup_start_time)
+        self.per_unit_storage_throughput = Resource.init_param(per_unit_storage_throughput)
+        self.backup_id = Resource.init_param(backup_id)
+        self.kms_key_id = Resource.init_param(kms_key_id)
+        self.file_system_id = Resource.init_param(file_system_id)
+        self.auto_import_policy = Resource.init_param(auto_import_policy)
+        self.drive_cache_type = Resource.init_param(drive_cache_type)
+        self.storage_type = Resource.init_param(storage_type)
 
     def _register_validators(self):
         self._add_validator(
```

Comment:
We talked about this and we reach the agreement that validators should keep receiving Parameter instances rather than just values in order to be able to reference them in the list of failed params.